chore: add api_keys_required decorator to Firecrawl Loader

a7m-1st opened the pull request:
## Description

Add api_keys_required decorator to Firecrawl Loader, instead of relying on catching the error lateron.

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have linked this PR to an issue using the Development section on the right sidebar or by adding `Fixes #issue-number` in the PR description (**required**)
- [x] I have checked if any dependencies need to be added or updated in `pyproject.toml` and `uv lock`
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation if needed:
- [ ] I have added examples if this is a new feature

If you are unsure about any of these, don't hesitate to ask. We are here to help!

Wendong-Fan reviewed (commented):
thanks @a7m-1st 's contribution! According to https://github.com/mendableai/firecrawl/blob/main/SELF_HOST.md#api-keys-for-sdk-usage
for self-deployed service the api key is optional

a7m-1st commented:
Oh okay I didn't realize it was intentional @Wendong-Fan. Then I have removed the additions and added a note instead from the given source to clarify for future contributors

GitHoobar commented:
Thanks @a7m-1st 
I think there's no actual need of this to explicitly mention in docstring.
